theano/sandbox/cuda/dnn.py

```diff
@@ -2186,6 +2186,460 @@ def method(self):
 #endif
         """
 
+class GpuDnnBatchNormBase(DnnBase):
+    """
+    Op for the cuDNN Batch Normalization.
+
+    Parameters
+    ----------
+    tensor_format
+        Always set this to 'bc01'.
+    mode
+        Normalization is performed per-activation for 'per-activation' (in this mode bnBias and bnScale tensor
+        dimensions are 1xCxHxW for 4d tensor) or Normalization is performed over N+spatial dimensions for 'spatial'
+        (in this mode bnBias, bnScale tensor dimensions are 1xCx1x1).
+    exp_avg_factor
+        Factor used in the moving average computation runningMean = newMean*factor + runningMean*(1-factor). It is
+        required that exponentialAverageFactor=1 is used for the very first call of a complete training cycle.
+    epsilon
+        Epsilon value used in the batch normalization formula. It is 1e-4 by default. Minimum allowed value
+        for cuDNN v.4 is 1e-5.
+
+    """
+
+    __props__ = ('tensor_format', 'mode', 'exp_avg_factor', 'epsilon')
+
+    def __init__(self, tensor_format, mode, exp_avg_factor=0.1, epsilon=1e-4):
+        if tensor_format != 'bc01':
+            raise ValueError(
+                "It was discovered that since December 2014, the "
+                "tensor_format parameter was ignored and the equivalent of "
+                "'bc01' is always used.  Since your code seems to be using "
+                "another value, this might have affected previous results "
+                "ran with this code.")
```

Comment:
who ignore it? Theano or CuDNN? Theano can't have ignore it as we didn't had it before. Just specify it in the message that it is CuDNN that ignored it and tell which version.